ci: Migrate slow_gradcheck to GHA

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #65730

This should close out the door on migrating all scheduled workflows we have for CircleCI

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

cc @ezyang @seemethere @malfet @pytorch/pytorch-dev-infra

Differential Revision: [D31225188](https://our.internmc.facebook.com/intern/diff/D31225188)

seemethere committed:
ci: Migrate slow_gradcheck to GHA

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on "ci: Migrate slow_gradcheck to GHA"


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on "ci: Migrate slow_gradcheck to GHA"


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on "ci: Migrate slow_gradcheck to GHA"


This should close out the door on migrating all scheduled workflows we have for CircleCI

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

cc @ezyang @seemethere @malfet @pytorch/pytorch-dev-infra

Differential Revision: [D31225188](https://our.internmc.facebook.com/intern/diff/D31225188)

[ghstack-poisoned]

albanD commented on .github/scripts/generate_ci_workflows.py:
```diff
@@ -469,6 +470,20 @@ def generate_workflow_file(self, workflow_template: jinja2.Template) -> None:
     ),
+    CIWorkflow(
+        arch="linux",
+        build_environment="periodic-pytorch-linux-xenial-cuda10.2-cudnn7-py3-gcc7-old-gradcheck",
```
nit: I realized after the fact that the label says "slow" while the build name says "old". It would be nice to fix that discrepancy and call everything "slow".
This might be a good time to make that change to avoid breaking things again in the future?

albanD reviewed (commented):
I'm not a specialist of the ciflow status but the change looks good (only the name thing below)
The next steps are going to be ? :
- delete the ci/slow-gradcheck label
- update the wiki

albanD commented:
Also, does this mean that you will need to opt in to ciflow to be able to run the slow gradcheck on a PR?
I use this label on a lot of OSS PR that add formula. Would that still be possible as ciflow is not enabled by default from what I understand.

seemethere commented:
> Also, does this mean that you will need to opt in to ciflow to be able to run the slow gradcheck on a PR? I use this label on a lot of OSS PR that add formula. Would that still be possible as ciflow is not enabled by default from what I understand.

In the near future everyone will be opted in to ciflow so this shouldn't be an issue

seemethere commented:
@pytorchbot ciflow rerun -l ciflow/cuda -l ciflow/slow

albanD commented:
Ho yes, I was just wondering if this will land in that interval? Like I don't want to slap the old label expecting it to work and actually nothing happening :)

seemethere commented:
> Ho yes, I was just wondering if this will land in that interval? Like I don't want to use the old label expecting it to work and actually nothing happening :)

I'll hold off merging this until we have that

seemethere commented:
@pytorchbot ciflow rerun -l ciflow/cuda -l ciflow/slow

seemethere committed:
Update on "ci: Migrate slow_gradcheck to GHA"


This should close out the door on migrating all scheduled workflows we have for CircleCI

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

cc ezyang seemethere malfet pytorch/pytorch-dev-infra

Differential Revision: [D31225188](https://our.internmc.facebook.com/intern/diff/D31225188)

[ghstack-poisoned]

seemethere commented on .github/scripts/generate_ci_workflows.py:
```diff
@@ -155,6 +156,7 @@ class CIWorkflow:
     only_run_smoke_tests_on_pull_request: bool = False
     num_test_shards_on_pull_request: int = -1
     distributed_test: bool = True
+    timeout_after: int = 240
```
Also added a variable timeout here with a default, if we think this is too long we can change this to be something else later on

seemethere committed:
Update on "ci: Migrate slow_gradcheck to GHA"


This should close out the door on migrating all scheduled workflows we have for CircleCI

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

cc ezyang seemethere malfet pytorch/pytorch-dev-infra

Differential Revision: [D31225188](https://our.internmc.facebook.com/intern/diff/D31225188)

[ghstack-poisoned]

seemethere committed:
Update on "ci: Migrate slow_gradcheck to GHA"


This should close out the door on migrating all scheduled workflows we have for CircleCI

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

cc ezyang seemethere malfet pytorch/pytorch-dev-infra

Differential Revision: [D31225188](https://our.internmc.facebook.com/intern/diff/D31225188)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31225188).

seemethere commented:
> > Ho yes, I was just wondering if this will land in that interval? Like I don't want to use the old label expecting it to work and actually nothing happening :)
> 
> I'll hold off merging this until we have that
> 
> related: [pytorch/probot#76](https://github.com/pytorch/probot/pull/76)

ciflow should be rolled out to everyone, going to go ahead and land this
